lib/shopify_api/bulk/query.ex

```diff
@@ -97,6 +98,51 @@ defmodule ShopifyAPI.Bulk.Query do
     end
   end
 
+  @spec stream_fetch(String.t() | {}) :: Enumerable.t()
+  def stream_fetch({:ok, url}), do: stream_fetch(url)
+  def stream_fetch({:error, _} = error), do: error
+
+  def stream_fetch(url) do
```

Comment:
Since further in we invoke a `receive` block, I don't _think_ this is safe to call from inside of a GenServer or similar OTP abstraction that manages it's own `receive` loop. Worth noting as documentation, perhaps?

Reply:
Good call, this whole function could use some documentation